lib/core/include/lsUtil.h

```diff
@@ -1,66 +1,82 @@
-/*** Copyright (c), The Regents of the University of California            ***
- *** For more information please refer to files in the COPYRIGHT directory ***/
-/* getUtil.h - Header for for getUtil.c */
+#ifndef IRODS_LS_UTIL_H
+#define IRODS_LS_UTIL_H
 
-#ifndef LS_UTIL_H__
-#define LS_UTIL_H__
+#include "rodsType.h"
 
-#include "rodsClient.h"
-#include "parseCommandLine.h"
-#include "rodsPath.h"
+struct RcComm;
+struct RodsEnvironment;
+struct RodsArguments;
+struct RodsPathInp;
+struct RodsPath;
+struct CollEnt;
+struct SpecColl;
+struct GenQueryOut;
 
 #ifdef __cplusplus
 extern "C" {
 #endif
-int
-lsUtil( rcComm_t *conn, rodsEnv *myRodsEnv, rodsArguments_t *myRodsArgs,
-        rodsPathInp_t *rodsPathInp );
-int
-lsDataObjUtil( rcComm_t *conn, rodsPath_t *srcPath,
-               rodsArguments_t *rodsArgs, genQueryInp_t *genQueryInp );
-int
-printLsStrShort( char *srcPath );
-int
-lsDataObjUtilLong( rcComm_t *conn, char *srcPath,
-                   rodsArguments_t *rodsArgs, genQueryInp_t *genQueryInp );
```

Comment:
I realize that this file is pretty specific to the iCommand `ils`, but is it bad to change the signature of an exposed function like this when it is not in the `irods::experimental` namespace? For this case it might not matter too much, just wondering how you felt about that in general.

Reply:
Yes, it is bad to do what I've done here. My first thought is that nothing except the icommands should be using these functions, but because they are already public, we can't enforce that. Therefore, it would be better to mark the old version of the functions in question as deprecated and add a new function that presents the new signature. The old function can be implemented in terms of the new one.

Reply:
I think I've restored the signature. Instead of marking things as deprecated, I chose to keep the API as they were and just ignore unused arguments. We can clean these up in 4.3.0+.